storage/range.go

```diff
@@ -2219,6 +2221,37 @@ func (r *Range) ChangeReplicas(changeType proto.ReplicaChangeType, replica proto
 	return nil
 }
 
+// updateRangeDescriptor returns a client.Call to execute a ConditionalPut
+// on the range descriptor.
```

Comment:
This is really where some of the detail from the PR description would help people down the road quickly determine why the conditional put is necessary. How about:

// The conditional put verifies that changes to the range descriptor are made
// in a well-defined order, preventing a scenario where a wayward replica which
// is no longer part of the original Raft group comes back online to form a splinter
// group with a node which was also a former replica, and hijacks the range
// descriptor.